chromadb/execution/executor/distributed.py

```diff
@@ -1,27 +1,207 @@
+            query = VectorQuery(
+                vectors=plan.knn.embeddings,
+                k=plan.knn.fetch,
+                allowed_ids=prefiltered_ids,
+                include_embeddings=plan.projection.embedding,
+                options=None,
+                request_version_context=plan.scan.version,
+            )
+            knns = self._vector_segment(plan.scan.collection).query_vectors(query)
+
+        ids = [[r["id"] for r in result] for result in knns]
+        embeddings = None
+        documents = None
+        uris = None
+        metadatas = None
+        distances = None
+        included = list()
+
+        if plan.projection.embedding:
+            embeddings = [[r["embedding"] for r in result] for result in knns]
+            included.append(IncludeEnum.embeddings)
+
+        if plan.projection.rank:
+            distances = [[r["distance"] for r in result] for result in knns]
+            included.append(IncludeEnum.distances)
+
+        if plan.projection.document or plan.projection.metadata or plan.projection.uri:
+            merged_ids = list(set([id for result in ids for id in result]))
+            hydrated_records = self._metadata_segment(
+                plan.scan.collection
+            ).get_metadata(
+                request_version_context=plan.scan.version,
+                where=None,
+                where_document=None,
+                ids=merged_ids,
+                limit=None,
+                offset=0,
+                include_metadata=True,
+            )
+            metadata_by_id = {r["id"]: r["metadata"] for r in hydrated_records}
+
+            if plan.projection.document:
+                documents = [
+                    [_doc(metadata_by_id.get(id, None)) for id in result]
+                    for result in ids
+                ]
+                included.append(IncludeEnum.documents)
+
+            if plan.projection.uri:
+                uris = [
+                    [_uri(metadata_by_id.get(id, None)) for id in result]
+                    for result in ids
+                ]
+                included.append(IncludeEnum.uris)
+
+            if plan.projection.metadata:
+                metadatas = [
+                    [_clean_metadata(metadata_by_id.get(id, None)) for id in result]
+                    for result in ids
+                ]
+                included.append(IncludeEnum.metadatas)
+
+        # TODO: Fix typing
+        return QueryResult(
+            ids=ids,
+            embeddings=embeddings,  # type: ignore[typeddict-item]
+            documents=documents,  # type: ignore[typeddict-item]
+            uris=uris,  # type: ignore[typeddict-item]
+            data=None,
+            metadatas=metadatas,  # type: ignore[typeddict-item]
+            distances=distances,
+            included=included,
+        )
+
+    def _metadata_segment(self, collection: Collection) -> GrpcMetadataSegment:
+        return self._manager.get_segment(collection.id, GrpcMetadataSegment)
```

Comment:
Did we consider how caching affects this / performance of not having it statically in the plan?